src/app/course-list/course-list.component.ts

```diff
@@ -23,11 +23,13 @@ export class CourseListComponent implements OnInit {
   deleteCourse(courseId: string) {
     this.courseService
       .deleteCourse(this.courses, courseId)
-      .subscribe(res => {
-        if (res) {
-          this.courses = res;
-        }
-      });
+      .subscribe(res => this.courses = res);
+  }
+
+  editCourse(courseId: string) {
+    this.courseService
+      .deleteCourse(this.courses, courseId)
```

Comment:
why do you want to delete course on edit? There should be navigation to edit page `router.navogate(['course', id])`